esp/smc/SMCLib/TpWrapper.cpp

```diff
@@ -1217,7 +1217,7 @@ void CTpWrapper::getGroupList(double espVersion, const char* kindReq, IArrayOf<I
             {
                 IPropertyTree &group = groups->query();
                 const char* kind = group.queryProp("@kind");
-                if (kindReq && *kindReq && !strieq(kindReq, kind))
+                if (!isEmptyString(kindReq) && !strisame(kindReq, kind))
```

Comment:
I'm not sure what the context/purpose/use-cases of this getGroupList call is, but do you ever want to return groups without a @kind? 
If not should be changed to skip them too.